test/base_test.go

```diff
@@ -0,0 +1,132 @@
+package test
+
+import (
+	"github.com/segmentio/kafka-go"
+	"io/ioutil"
+	"context"
+	"log"
+	"net/http"
+	"testing"
+	"time"
+)
+
+/* TODO: current version doesn't use dynamic port allocation due to Canary not running properly on Kafka mapped to any other port than default  9092
++ what is later propagated as series of unwanted behaviours, for example not having open port on 9092 ends in failed attempt to connect to this port despite other port being set in configuration
+*/
```

Comment:
I don't see this problem. I have just run a local Kafka cluster with one broker listening on port 9093 (so not 9092) and configuring the canary with the KAFKA_BOOTSTRAP_SERVER env var at localhost:9093 and it just connects and works.

Reply:
in my case (kafka running on port 9093, env variable set:  KAFKA_BOOTSTRAP_SERVERS=localhost:9093, and port 9092 closed ) it ends up like this: 
```
E0707 13:42:01.424011   69927 topic.go:104] Error describing cluster: dial tcp 127.0.0.1:9092: connect: connection refused
F0707 13:42:01.424028   69927 canary_manager.go:76] Error starting manager: dial tcp 127.0.0.1:9092: connect: connection refused
```

and in same situation as above but with kafka running on both 9092 and 9093 problem appears only after 2 and a half minute:
```
Error joining the consumer group: Maximum number of attempts exceeded
exit status 255
```

Reply:
when the canary starts, it prints the configuration it's getting from the env vars. Checks what's your configuration and if it's not what you expect, you have some problem passing env vars but it can't be canary related.

Reply:
configuration is according to expectation, it is printed correctly.  (For upper example )
```
Starting Strimzi canary tool with config: {BootstrapServers:localhost:9093, ...
```

Reply:
can you provide a cleaner explanation and steps to reproduce what you are seeing please?